src/renderers/dom/client/ReactMount.js

```diff
@@ -109,13 +121,75 @@ function mountComponentIntoNode(
     console.time(markerName);
   }
 
-  var markup = ReactReconciler.mountComponent(
-    wrapperInstance,
-    transaction,
-    null,
-    ReactDOMContainerInfo(wrapperInstance, container),
-    context
-  );
+  try {
+    var markup = ReactReconciler.mountComponent(
+      wrapperInstance,
+      transaction,
+      null,
+      ReactDOMContainerInfo(wrapperInstance, container),
+      context,
+      shouldReuseMarkup ? container.firstChild : undefined
+    );
+  } catch (e) {
```

Comment:
This error handling had to move over from `_mountImageIntoNode` because now a markup mismatch results in a thrown Error from `mountComponent`.